source/frontend/src/hooks/useBcAssessmentLayer.ts

```diff
@@ -0,0 +1,393 @@
+import { IResponseWrapper, useApiRequestWrapper } from './pims-api/useApiRequestWrapper';
+import useKeycloakWrapper from './useKeycloakWrapper';
+import { useModalContext } from './useModalContext';
+
+/**
+ * API wrapper to centralize all AJAX requests to WFS endpoints on the BC assessment layer.
+ * @returns Object containing functions to make requests to the WFS layer.
+ * Note: according to https://catalogue.data.gov.bc.ca/dataset/parcelmap-bc-parcel-fabric/resource/959af382-fb31-4f57-b8ea-e6dcb6ce2e0b
+ */
+export const useBcAssessmentLayer = (
+  url: string,
+  names: { [key: string]: string },
+): {
+  getSummaryWrapper: IResponseWrapper<
+    (pid: string) => Promise<AxiosResponse<IBcAssessmentSummary>>
+  >;
+} => {
+  const keycloak = useKeycloakWrapper();
+  const logout = keycloak.obj.logout;
+  const { setModalContent, setDisplayModal } = useModalContext();
+  useEffect(() => {
+    setModalContent({
+      title: 'SiteMinder Session Expired',
+      message:
+        'Your SiteMinder Session has expired. In order to access BC Assessment data, you will need to logout and log back in to the application.',
+      okButtonText: 'Log out',
+      cancelButtonText: 'Continue working',
+      handleOk: () => {
+        logout();
+      },
+      handleCancel: () => {
+        setDisplayModal(false);
+      },
+    });
+  }, [setModalContent, logout, setDisplayModal]);
+
+  const getLegalDescriptionsWrapper = useWfsLayer(
+    url,
+    {
+      name: names['LEGAL_DESCRIPTION'],
+      withCredentials: true,
+    },
+    { throwError: true },
+  );
+  const getAddressesWrapper = useWfsLayer(url, {
+    name: names['ADDRESS'],
+    withCredentials: true,
+  });
+  const getValuesWrapper = useWfsLayer(url, {
+    name: names['VALUE'],
+    withCredentials: true,
+  });
+  const getChargesWrapper = useWfsLayer(url, {
+    name: names['CHARGE'],
+    withCredentials: true,
+  });
+  const getFolioDescriptionsWrapper = useWfsLayer(url, {
+    name: names['FOLIO_DESCRIPTION'],
+    withCredentials: true,
+  });
+  const getSalesWrapper = useWfsLayer(url, {
+    name: names['SALE'],
+    withCredentials: true,
+  });
+  const getSales = getSalesWrapper.execute;
+  const getFolioDescriptions = getFolioDescriptionsWrapper.execute;
+  const getCharges = getChargesWrapper.execute;
+  const getValues = getValuesWrapper.execute;
+  const getAddresses = getAddressesWrapper.execute;
+  const getLegalDescriptions = getLegalDescriptionsWrapper.execute;
+
+  const getSummary = useCallback(
+    async (pid: string): Promise<AxiosResponse<IBcAssessmentSummary>> => {
+      const parsedPid = pidParser(pid);
+      if (parsedPid === undefined) {
+        throw Error(`Unable to parse PID, invalid format: ${pid}`);
+      }
+      if (process.env.NODE_ENV === 'development') {
+        return {
+          data: mockBcAssessmentSummary,
+          status: 200,
+          statusText: 'Success',
+          headers: {},
+          config: {},
+        };
+      }
```

Comment:
this is certainly a new pattern for local development - I gather we didn't want to setup a local WFS layer in geoserver for this?

Reply:
we don't have the underlying data in this case.

Reply:
Logged https://jira.th.gov.bc.ca/browse/PSP-4852 as a follow up, should be possible to request access to the underlying layers.